bundle/src/main/java/com/adobe/acs/commons/packaging/impl/AssetPackagerServletImpl.java

```diff
@@ -0,0 +1,281 @@
+
+    @Reference
+    private PackageHelper packageHelper;
+
+    private String customPrefix;
+    private List<Pattern> pageExclusionPatterns;
+    private List<Pattern> assetExclusionPatterns;
+    private final List<String> excludedPages = new ArrayList<>();
+
+    @Override
+    public final void doPost(final SlingHttpServletRequest request,
+                             final SlingHttpServletResponse response) throws IOException {
+
+        final ResourceResolver resourceResolver = request.getResourceResolver();
+        final boolean preview = Boolean.parseBoolean(request.getParameter("preview"));
+
+        log.debug("Preview mode: {}", preview);
+
+        final ValueMap properties = this.getProperties(request);
+
+        this.customPrefix = properties.get(PN_ASSET_PREFIX, String.class);
+        this.pageExclusionPatterns = generatePatterns(properties.get(PN_PAGE_EXCLUSIONS, new String[]{}));
+        this.assetExclusionPatterns = generatePatterns(properties.get(PN_ASSET_EXCLUSIONS, new String[]{}));
+
+        try {
+            final String pagePath = properties.get(PN_PAGE_PATH, String.class);
+            final List<PathFilterSet> paths;
+            if (StringUtils.isBlank(pagePath)) {
+                paths = Collections.emptyList();
+            } else {
+                paths = this.findAssetPaths(resourceResolver, pagePath);
+            }
+            if (!properties.get(PN_EXCLUDE_PAGES, false) && paths.size() > 0) {
+                PathFilterSet pageFilter = new PathFilterSet(pagePath);
+                if (excludedPages.size() > 0) {
+                    for (String excludedPath : excludedPages) {
+                        pageFilter.addExclude(new DefaultPathFilter(excludedPath));
+                    }
+                }
+                paths.add(0, pageFilter);
+            }
+
+            doPackaging(request, response, preview, properties, paths);
+        } catch (RepositoryException ex) {
+            log.error("Repository error while creating Asset Package", ex);
+            response.getWriter().print(packageHelper.getErrorJSON(ex.getMessage()));
+        } catch (IOException ex) {
+            log.error("IO error while creating Asset Package", ex);
+            response.getWriter().print(packageHelper.getErrorJSON(ex.getMessage()));
+        }
+    }
+
+    /**
+     * Gets the properties saved to the Asset Packager Page's jcr:content node.
+     *
+     * @param request The request obj
+     * @return A ValueMap representing the properties
+     */
+    private ValueMap getProperties(final SlingHttpServletRequest request) {
+        if (request.getResource().getChild("configuration") == null) {
+            log.warn("Asset Packager Configuration node could not be found for: {}", request.getResource());
+            return new ValueMapDecorator(new HashMap<>());
+        } else {
+            return request.getResource().getChild("configuration").adaptTo(ValueMap.class);
```

Comment:
`.adaptTo(ValueMap.class` can instead be `getValueMap()` (optional refactor)

Reply:
+1 - generally speaking, `.getValueMap()` is preferred as it guarantees a non-null (Empty value map) return value, whereas with `.adaptTo(ValueMap.class)` requires a null check before accessing.

Reply:
Addressed!